lib/Doctrine/Migrations/Metadata/AvailableMigrationsSet.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Metadata;
+
+use Countable;
+use Doctrine\Migrations\Exception\MigrationNotAvailable;
+use Doctrine\Migrations\Version\Version;
+use function array_values;
+use function count;
+
+/**
+ * The class AvailableMigrationsSet represents a non sorted set of available migrations.
+ * The migrations in this set might be already executed or not.
+ */
+final class AvailableMigrationsSet implements Countable
+{
+    /** @var AvailableMigration[] */
```

Comment:
Since we have `array_values`…
```suggestion
    /** @var list<AvailableMigration> */
```

Reply:
Is this compatible with common Ides?

Reply:
It's compatible with vim :P

Reply:
Alternative suggested [here](https://twitter.com/Ocramius/status/1258360055151439872):

```suggestion
    /**
     * @var AvailableMigration[]
     * @phpstan-var AvailableMigration[]
     * @psalm-var AvailableMigration[]
     */
```

Not that the second one is superflous: both tools read each other's annotations

Reply:
I personally do not like this "tool" specific annotations... if a third tool will emerge will need to add them too... i suggest to leave `AvailableMigration[]` as it is